server/controllers/UserController.js

```diff
@@ -0,0 +1,220 @@
+import jwt from 'jsonwebtoken';
+import model from '../models';
+
+const User = model.User;
+const Documents = model.Document;
+const secret = 'secret';
+
+
+const UserController = {
+  login(request, response) {
+    if (request.body.loginId === '' || request.body.password === '') {
+      return response.status(400).json({
+        status: 'Bad Request',
+        message: 'Fields cannot be empty'
+      });
+    }
+    return User
+      .findOne({
+        where: {
+          $or: [{
+            username: request.body.loginId
+          }, {
+            email: request.body.loginId
+          }]
+        }
+      })
+      .then((user) => {
+        if (!user) {
+          return response.status(404).send({
+            status: 404,
+            message: 'User does not exist'
+          });
+        } else if (user.validPassword(request.body.password)) {
+          const userData = {
+            id: user.id,
+            name: `${user.firstname} ${user.lastname}`,
+            username: user.username,
+            email: user.email,
+            phone: user.phone,
+            roleId: user.roleId
+          };
+          const token = jwt.sign({
+            id: user.id,
+            roleId: user.roleId,
+            expiresIn: '1hr'
+          }, secret);
+          return response.status(200).send({
+            status: 200,
+            userData,
+            message: `Welcome ${userData.name}`,
+            token
+          });
+        }
+        return response.status(400).json({
+          status: 400,
+          message: 'Could not sign you in. Kindly check your login details'
+        });
+      })
+      .catch(error => response.status(400).send(error));
+  },
+  logout(request, response) {
+    response.setHeader['x-access-token'] = ' ';
+    response.status(200)
+      .json({
+        success: true,
+        message: 'User logged out'
+      });
+  },
+  create(request, response) {
+    User
+      .findOne({
```

Comment:
Adding Unique to the sequlize model removes the redundancy of finding before creating a user